src/istgt.c

```diff
@@ -317,33 +317,42 @@ istgt_add_portal_group(ISTGT_Ptr istgt, CF_SECTION *sp, int *pgp_idx)
 static int
 istgt_pg_match_all(PORTAL_GROUP *pgp, CF_SECTION *sp)
 {
-	char *label, *portal, *host, *port;
+	char *label = NULL, *portal = NULL, *host = NULL, *port = NULL;
 	int que;
 	int rc;
+	int ret = 0;
 	int i;
 
 	for (i = 0; i < pgp->nportals; i++) {
 		label = istgt_get_nmval(sp, "Portal", i, 0);
 		portal = istgt_get_nmval(sp, "Portal", i, 1);
 		if (label == NULL || portal == NULL)
-			return 0;
+			goto no_match;
 		rc = istgt_parse_portal(portal, &host, &port, &que);
 		if (rc < 0)
-			return 0;
+			goto no_match;
 		if (strcmp(pgp->portals[i]->label, label) != 0)
-			return 0;
+			goto no_match;
 		if (strcmp(pgp->portals[i]->host, host) != 0)
-			return 0;
+			goto no_match;
 		if (strcmp(pgp->portals[i]->port, port) != 0)
-			return 0;
+			goto no_match;
 		if (pgp->portals[i]->que != que)
-			return 0;
+			goto no_match;
 	}
+
 	label = istgt_get_nmval(sp, "Portal", i, 0);
 	portal = istgt_get_nmval(sp, "Portal", i, 1);
 	if (label != NULL || portal != NULL)
-		return 0;
-	return 1;
+		goto no_match;
+	ret = 1;
+
+no_match:
+	if (port)
```

Comment:
'free'ing here doesn't look correct, as, this memory area may be still being pointed by data structures in 'istgt->config->section'

Reply:
In `istgt_parse_portal`, we are allocating memory for `port` and copy port number from istgt->config->section for only comparison purpose. So, we can free `port` and `host` here.